### PR TITLE
(Draft) Testing a few backlogged Zero variants (+ a few others)

### PR DESCRIPTION
**Pull request by Logo121**
Starting this because the major number is going to change anyway:
image

This covers the proposed zero variants suggested in:
* #1307:
  * Slash cutout with same slope as the default slashed 0 (https://github.com/be5invis/Iosevka/issues/1307#issue-1138378807)
  * Slash cutout with steeper slope (https://github.com/be5invis/Iosevka/issues/1307#issuecomment-1052195735)
  * Top-right cutout, like the license plate font (https://github.com/be5invis/Iosevka/issues/1307#issuecomment-1053588047)
  * Vertical cutout, like stencil font (https://github.com/be5invis/Iosevka/issues/1307#issuecomment-1553649962)
* #1509
  * Slash with steeper slope (from (0, 0) to (Width, top), I assume)
    * This is aligned with the steeper-slope cutout above
* #1678 
  * Broken vertical bar
  * Broken Slash, like Fixedsys, as I've mentioned and demonstrated (https://github.com/be5invis/Iosevka/issues/1678#issuecomment-1518214429)
    * The slope is aligned with that of Empty Set, because there would be very little remaining if the other slopes are used

Plus their respective Oval/onum forms. The config has also been modified to support the "broken" shapes.

This is just for testing now, and I don't think *all* of these (especially those raised by me) are to be added anyway. For this reason, I also didn't make the reversed slash equivalent, or edit the variant table. These are all encoded in the PUA (U+EF00...U+EF1B) for now.

This is how it looks like zoomed out: image

I'm putting this here to know what should be done next, in particular:
* Which to keep, or even what to add (e.g. reversed variants)
* What optimizations/fixes should be done
* How to order them with the existing zero variants

### CODE REVIEW

**Comment by wimstefan** [comment]
Thank you so much for all the hard work you put into it!
It would be simply fantastic to see my cautious request being realized after such a long time 😃

File: font-src/glyphs/number/0.ptl [hunk 1]
```diff
@@ -31,16 +34,65 @@ glyph-block Digits-Zero : begin
...
+	define [CutoutSlash top] : dispiro
+		flat 0 (top * (1 - 0.77)) [widths.center (top * 0.125)]
+		curl Width (top * 0.77)
+
+	define [CutoutTallSlash top] : dispiro
+		flat 0 0 [widths.center (top * 0.125)]
+		curl Width top
+
+	define [CutoutVerticalBar top] : dispiro
+		flat Middle (top - O) [widths.center (Width * 0.125)]
```
Comment (be5invis) [inline font-src/glyphs/number/0.ptl:60]: Change the stroke width after (`widths.center`) to things like
`[Math.min (0.4 * CircleInnerWidth) [AdviceStroke 4]]`

File: font-src/glyphs/number/0.ptl [hunk 2]
```diff
@@ -31,16 +34,65 @@ glyph-block Digits-Zero : begin
...
+			include : difference zero overlay
+			include zero
+
+
+	define [CutoutSlash top] : dispiro
+		flat 0 (top * (1 - 0.77)) [widths.center (top * 0.125)]
+		curl Width (top * 0.77)
+
+	define [CutoutTallSlash top] : dispiro
+		flat 0 0 [widths.center (top * 0.125)]
```
Comment (be5invis) [inline font-src/glyphs/number/0.ptl:56]: Change the stroke width after (`widths.center`) to things like
`[Math.min (0.4 * CircleInnerWidth) (top * 0.15) [AdviceStroke 3]]`

Adjust the parameters to make the SW being proper under all the weights.

File: font-src/glyphs/number/0.ptl [hunk 3]
```diff
@@ -31,16 +34,65 @@ glyph-block Digits-Zero : begin
...
+			include overlay
+
+	define [Cutout zero overlay fnMask height] : glyph-proc
+		if overlay
+			include : difference zero overlay
+			include zero
+
+
+	define [CutoutSlash top] : dispiro
+		flat 0 (top * (1 - 0.77)) [widths.center (top * 0.125)]
```
Comment (be5invis) [inline font-src/glyphs/number/0.ptl:52]: Change the stroke width after (`widths.center`) to things like
`[Math.min (0.4 * CircleInnerWidth) (top * 0.15) [AdviceStroke 3]]`

Adjust the parameters to make the SW being proper under all the weights.

File: font-src/glyphs/number/0.ptl (hunk 1 above)
Comment (be5invis) [reply font-src/glyphs/number/0.ptl:60]: Adjust the parameters to make the SW being proper under all the weights.

**Review by be5invis — CHANGES_REQUESTED** [review]

**Comment by be5invis** [comment]
And well, you can keep all the variants (modify `variants.toml` to add them under `[prime.zero.variants]`). Since V23 will be a major version change you can freely reorder the variants' rank.

**Comment by Lieutenant-L-T-Smash** [comment]
+1 to the steeper slash. If anything makes it through to final, it must be that one. Looking at this chart and being able to compare, it's surprising even to me how much difference that makes in making it distinguishable from the 8 under astigmatic vision.

The vertical cutout just looks like a pair of parentheses to me.

**Comment by Logo121** [comment]
Ok, I tried making the gap vary a bit more, though if I just use AdviceStroke without a lower bound, the gap will be almost non-existent in Thin (and Extralight to some extent): image

I tried to keep the effect that the gaps are still discernible when zoomed out (or at around 8pt), at least for me:
imageimage

And this is what it looks like up close (at around 24pt):
image image

Not sure if it's better now.

**Pull request merged**